linopy/solvers.py

```diff
@@ -873,24 +932,28 @@ def run_mosek(
         "solsta.dual_infeas_cer": "infeasible_or_unbounded",
     }
 
-    if io_api is not None and io_api not in ["lp", "mps"]:
+    if io_api is not None and io_api not in ["lp", "mps", "direct"]:
         raise ValueError(
             "Keyword argument `io_api` has to be one of `lp`, `mps` or None"
         )
 
     problem_fn = model.to_file(problem_fn)
 
-    problem_fn = maybe_convert_path(problem_fn)
+    if io_api != "direct" and io_api is not None:
+        problem_fn = maybe_convert_path(problem_fn)
     log_fn = maybe_convert_path(log_fn)
-    warmstart_fn = maybe_convert_path(warmstart_fn)
-    basis_fn = maybe_convert_path(basis_fn)
+    # warmstart_fn = maybe_convert_path(warmstart_fn)
+    # basis_fn = maybe_convert_path(basis_fn)
```

Comment:
this should still be fine being uncommented or?

Reply:
Done